src/resources/views/crud/fields/image.blade.php

```diff
@@ -1,13 +1,23 @@
 @php
-    $prefix = isset($field['prefix']) ? $field['prefix'] : '';
+    $prefix = $field['prefix'] ?? '';
+    $disk = $field['disk'] ?? null;
     $value = old(square_brackets_to_dots($field['name'])) ?? $field['value'] ?? $field['default'] ?? '';
-    $value = $value
-        ? preg_match('/^data\:image\//', $value)
-            ? $value
-            : (isset($field['disk'])
-                ? Storage::disk($field['disk'])->url($prefix.$value)
-                : url($prefix.$value))
-        :''; // if validation failed, tha value will be base64, so no need to create a URL for it
+
+    if (! function_exists('getDiskUrl')) {
+        function getDiskUrl($disk, $path) {
+            try {
+                // make sure the value don't have disk base path on it
+                $origin = substr(Storage::disk($disk)->url('/'), 0, -1);
```

Comment:
Brief explanation on this;
```php
$origin = substr(Storage::disk($disk)->url('/'), 0, -1);
```
That is the same as;
```php
$origin = Storage::disk($disk)->url('');
```
But the second one doesn't work on `S3` disks because they require the path length to be `>= 1`.
Hence the complicated form.

Reply:
Didn't know about this requirement in s3. Thanks for checking @promatik . Google Storage allow `...->url('')` and works fine (i had tested).